packages/spacecat-shared-data-access/src/models/site/site.schema.js

```diff
@@ -69,6 +69,11 @@ const schema = new SchemaBuilder(Site, SiteCollection)
     default: {},
     validate: (value) => isObject(value),
   })
+  .addAttribute('siteCompetitors', {
+    type: 'list',
+    default: [],
+    validate: (value) => Array.isArray(value),
```

Comment:
if the attribute is not required, we should not check for array unless a value is present.